code/script.js

```diff
@@ -229,97 +232,162 @@ const setSecret = () => {
     }
   }
+  else if (category === 'hair'){
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
+    if (keep) {
+       alert( 
+         `Yes, the person has ${value} hair! Keep all people with ${value} hair!`
+       )
+       charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+    } 
+    else {
+       alert( 
+        `No, the person doesnt have ${value} hair! Remove all people with ${value} hair`
+       )
+       charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+    }
+  }
 
-  // Invoke a function to redraw the board with the remaining people.
+  else if (category === 'eyes'){
+    if (keep) {
+      alert(
+        `Yes, the person has ${value} eyes! Keep all the people with ${value} eyes!`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person[category] == value)
+    }
+    else {
+      alert( 
+       `No, the person doesnt have ${value} eyes! Remove all people with ${value} eyes!`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+  }
+  }
+  generateBoard()
 }
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
   // store the interaction from the player in a variable.
+  const confirming = confirm(`Do you really want to guess ${personToConfirm}?`)
+  if (confirming === true) {
+checkMyGuess(personToConfirm)
+  }
+  
+  else {
+    alert (`You Wuss!`)
+  }
   // remember the confirm() ?
   // If the player wants to guess, invoke the checkMyGuess function.
 }
 
 // If you confirm, this function is invoked
-const checkMyGuess = (personToCheck) => {
+const checkMyGuess = (personToConfirm) => {
   // 1. Check if the personToCheck is the same as the secret person's name
+  if(secret.name === personToConfirm){
+    winOrLose.style.display ='block'
+    board.style.display = 'none'
+  
   // 2. Set a Message to show in the win or lose section accordingly
+  winOrLoseText.innerHTML = `<h1>You Won You Lucky Bastard!</h1>`
+}
   // 3. Show the win or lose section
-  // 4. Hide the game board
+  else {
+    winOrLose.style.display = 'block'
+    board.style.display = 'none'
+
+    winOrLoseText.innerHTML = `<h1> Sorry Dude, You Suck. Do You Wanna Try Again?</h1>`
+  }
 }
+  // 4. Hide the game board
+const playAgain = () => {
+  winOrLose.style.display = 'none'
+  board.style.display = 'flex'
```

Comment:
If you wanted to shorten your code, the info within this function could have been included inside the start-function instead.